serveradmin/common/static/js/serveradmin.js

```diff
@@ -7,23 +7,58 @@ $(document).ready(function() {
     // This is our progress spinner we can call e.g. spinner.enable() from
     // everywhere when ever we need it for example when doing long running
     // ajax requests for the servershell search.
-    spinner = {
+    window.spinner = {
+        _id: null,
+        _running: 0,
+        _spinner: $('#spinner'),
+        _counter: $('#spinner-counter'),
+        _value: $('#spinner-counter-value'),
+        _toggle_css: function() {
+            if (this._spinner.css('display') === 'none') {
+                this._spinner.show();
+                this._counter.show();
+            }
+
+            if (this._spinner.hasClass('text-success')) {
+                this._spinner.removeClass('text-success');
+                this._spinner.addClass('text-secondary');
+
+                this._counter.removeClass('text-success');
+                this._counter.addClass('text-secondary');
+
+                this._spinner.css('animation-play-state', 'paused');
+            }
+            else {
+                this._spinner.removeClass('text-secondary');
+                this._spinner.addClass('text-success');
+
+                this._counter.removeClass('text-secondary');
+                this._counter.addClass('text-success');
+
+                this._spinner.css('animation-play-state', 'running');
+            }
+        },
         enable: function () {
-            $('#spinner').show();
-            $('#spinner-counter').show();
-
-            let update_spinner = function() {
-                let element = $('#spinner-counter-time');
-                element.html(Number.parseInt(element.html()) + 1);
-                setTimeout(update_spinner, 1);
-            };
-            setTimeout(update_spinner, 1);
+            this._running++;
+            if (this._running > 1) {
+                return;
+            }
+
+            this._value.html(0);
+            this._toggle_css();
+
+            this._id = setInterval(function () {
+                spinner._value.html(Number.parseInt(spinner._value.html()) + 1);
+            }, 1);
         },
         disable: function () {
-            $('#spinner').hide();
-            $('#spinner-counter').hide();
-            $('#spinner-counter-time').html(0);
-            $('#spinner-counter-unit').html('ms');
+            this._running--;
+
+            if (this._running === 0) {
```

Comment:
Ha, I was writing a lot of stuff about a possible Race Condition and just read again that you do not need to worry about that in Javascript in browsers. There is no sort of concurrency in Javascript in browsers. Just leaving it again for a "nice to know".

Reply:
Thanks. Grateful for any explanation and knowledge. I never really "learned" Javascript and its odds.